cliapi/providers/azure.py

```diff
@@ -0,0 +1,66 @@
+from cliapi.cliapi_lib import Provider, cliapi_compile
+from cliapi.what_cloud import determine_provider
+
+import json
+import os
+import uuid
+import urllib.request, urllib.error, urllib.parse
+import xml.etree.ElementTree as ET
+
+try:
+    # ----- raises an error on import if not running in a "proper" cloud...
+    cp = determine_provider()
+    if cp != 'azure':
+        raise Exception("not a valid provider")
+except Exception as e:
+    raise e
+
+provider = Provider()
+
+scoops = {
+    'instance-name': "['meta_data']['compute']['name']",
+    'mac': "['meta_data']['network']['interface'][0]['macAddress']",
+    'location': "['meta_data']['compute']['location']",
+    'external-ip': "['meta_data']['network']['interface'][0]"
+                   "['ipv4']['ipAddress'][0]['publicIpAddress']",
+    'internal-ip': "['meta_data']['network']['interface'][0]"
+                   "['ipv4']['ipAddress'][0]['privateIpAddress']",
+    'cloud-service': "['get_cloud_service']"
+}
+
+help = {
+    'instance-name': 'name of instance',
+    'location': 'region location',
+    'mac': 'the MAC address for this interface',
+    'api_version': 'azure metadata api version',
+    'cloud-service': 'what'
+}
```

Comment:
There is no need for long help for the metadata tools. This may be different for other use cases of the framework. Thus the framework needs to handle a potentially empty help{} dictionary. One option to deal with this would be to make every plugin implement a generate_help() method. Then plugins that need long help can return a hard coded dict something along the lines of the above. For the cloud metadata tool we would simply use the keys of the scoops.